java/ql/lib/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -182,6 +204,695 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ".map;ListOrderedMap;true;setValue;;;Argument[1];MapValue of Argument[-1];value",
+          ".map;ListOrderedMap;true;put;;;Argument[1];MapKey of Argument[-1];value",
+          ".map;ListOrderedMap;true;put;;;Argument[2];MapValue of Argument[-1];value",
+          ".map;ListOrderedMap;true;remove;(int);;MapValue of Argument[0];ReturnValue;value",
+          ".map;ListOrderedMap;true;asList;;;MapKey of Argument[1];Element of ReturnValue;value",
+          ".map;LRUMap;true;LRUMap;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;LRUMap;true;LRUMap;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;LRUMap;true;LRUMap;(Map,boolean);;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".map;LRUMap;true;LRUMap;(Map,boolean);;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".map;LRUMap;true;get;(Object,boolean);;MapValue of Argument[0];ReturnValue;value",
+          ".map;MultiKeyMap;true;multiKeyMap;;;MapKey of Argument[0];MapKey of ReturnValue;value",
+          ".map;MultiKeyMap;true;multiKeyMap;;;MapValue of Argument[0];MapValue of ReturnValue;value",
+          ".map;MultiKeyMap;true;get;;;MapValue of Argument[-1];ReturnValue;value",
+          ".map;MultiKeyMap;true;put;;;MapValue of Argument[-1];ReturnValue;value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object);;Argument[0..1];Element of MapKey of Argument[-1];value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object,Object);;Argument[0..2];Element of MapKey of Argument[-1];value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object,Object,Object);;Argument[0..3];Element of MapKey of Argument[-1];value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object,Object,Object,Object);;Argument[0..4];Element of MapKey of Argument[-1];value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object);;Argument[2];MapValue of Argument[-1];value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object,Object);;Argument[3];MapValue of Argument[-1];value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object,Object,Object);;Argument[4];MapValue of Argument[-1];value",
+          ".map;MultiKeyMap;true;put;(Object,Object,Object,Object,Object,Object);;Argument[5];MapValue of Argument[-1];value",
+          ".map;MultiKeyMap;true;removeMultiKey;;;MapValue of Argument[-1];ReturnValue;value",
+          ".map;MultiValueMap;true;multiValueMap;;;MapKey of Argument[0];MapKey of ReturnValue;value",
+          ".map;MultiValueMap;true;multiValueMap;;;Element of MapValue of Argument[0];Element of MapValue of ReturnValue;value",
+          ".map;MultiValueMap;true;getCollection;;;Element of MapValue of Argument[-1];Element of ReturnValue;value",
+          ".map;MultiValueMap;true;putAll;;;Argument[0];MapKey of Argument[-1];value",
+          ".map;MultiValueMap;true;putAll;;;Element of Argument[1];Element of MapValue of Argument[-1];value",
```

Comment:
```suggestion
          ".map;MultiValueMap;true;putAll;;;Element of Argument[0];Element of MapValue of Argument[-1];value",
```

Reply:
I'm modelling the `putAll(K key, Collection<V> values)` method here, not `putAll(Map<? extends K,?> map)` (which is inherited from `Map`. I'll explicitly give the signature to make this clearer.